### PR TITLE
WebSocket: Add simple websocket implementation

### PR DESCRIPTION
**Pull request by TiomanGally**

### CODE REVIEW

File: examples/websocket/src/main/kotlin/cnt/websocket/LectureChatWebSocketServer.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+
+    @OnMessage
+    fun onMessage(message: Message, @PathParam("username") username: String) {
+        broadcast(">> ${message.sendAt} - ${username}: ${message.message}", username)
+    }
+
+    private fun broadcast(message: String, username: String? = null) {
+        sessions.forEach { (_, session) ->
```
Comment (Alexander-Miller) [inline examples/websocket/src/main/kotlin/cnt/websocket/LectureChatWebSocketServer.kt:63]: `sessions.values.forEach` might looks nicer.

File: examples/websocket/src/main/kotlin/cnt/websocket/Message.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package cnt.websocket
+
+import java.time.LocalDateTime
+
+@NoArgConstructor
+data class Message(
+    val message: String,
+    val sender: String? = null,
+    val sendAt: LocalDateTime = LocalDateTime.now()
```
Comment (Alexander-Miller) [inline examples/websocket/src/main/kotlin/cnt/websocket/Message.kt:9]: sent

File: examples/websocket/src/test/kotlin/cnt/websocket/LectureChatWebSocketServerTest.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        }
+
+        companion object {
+            private val message = LinkedBlockingDeque<Message>()
+
+            fun startRecordingMessage() {
+                message.clear()
+            }
+
+            fun stopRecordingMessages(expectedElements: Int): List<Message?> {
```
Comment (Alexander-Miller) [inline examples/websocket/src/test/kotlin/cnt/websocket/LectureChatWebSocketServerTest.kt:140]: The naming and implementation are confusing. This functions doesn't just stop. It also tries to read the given number of messages, and will presumably throw due to timeouts on waiting for the right number to arrive. So what's really happening is it tries to return some recorded data, implicitly asserting that all requested messages exist.

File: examples/websocket/src/main/kotlin/cnt/websocket/MessageEnDecoder.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import java.io.Writer
+import javax.websocket.Decoder
+import javax.websocket.Encoder
+import javax.websocket.EndpointConfig
+
+
+class MessageEnDecoder
+    : Encoder.TextStream<Message>, Decoder.TextStream<Message> {
+
+    private val objectMapper = ObjectMapper().apply {
```
[the remark this answers is not in the document]
Comment (TiomanGally) [reply examples/websocket/src/main/kotlin/cnt/websocket/MessageEnDecoder.kt:17]: Was not able to find anything. Also IntelliJ can not find any other implementation besides mine.

Comment (slu-it) [inline examples/websocket/src/main/kotlin/cnt/websocket/MessageEnDecoder.kt:17]: `ObjectMapper` uses a fluent syntax for high level configuration, so the `apply` should not be necessary.

```
ObjectMapper()
    .registerModule(..)
    . setDefault...
```

**Pull request merged**